src/sisl/io/vasp/stdout.py

```diff
@@ -101,49 +101,62 @@ def read_energy(self, all=False):
-            line = next(itt)
-            E = PropertyDict()
-            while "----" not in line:
-                key, *v = line.split()
-                if key == "PAW":
-                    E[f"{name_conv[key]}1"] = float(v[-2])
-                    E[f"{name_conv[key]}2"] = float(v[-1])
-                else:
-                    E[name_conv[key]] = float(v[-1])
-                line = next(itt)
-            line = next(itt)
-            E["free"] = float(line.split()[-2])
-            next(itt)
-            line = next(itt)
+        # read the energy tables
+        f = self.step_to("Free energy of the ion-electron system", allow_reread=False)[0]
+        if not f:
+            return None
+        self.readline() # -----
+        line = self.readline()
+        E = PropertyDict()
+        while "----" not in line:
+            key, *v = line.split()
+            if key == "PAW":
+                E[f"{name_conv[key]}1"] = float(v[-2])
+                E[f"{name_conv[key]}2"] = float(v[-1])
+            else:
+                E[name_conv[key]] = float(v[-1])
+            line = self.readline()
+        line = self.readline()
+        E.free = float(line.split()[-2])
+        self.readline()
+        line = self.readline()
+        v = line.split()
+        E.total = float(v[4])
+        E.sigma0 = float(v[-1])
+        print(E)
+        return E
+
+    @SileBinder()
+    @sile_fh_open()
+    def read_trajectory(self):
+        f = self.step_to("BASIS-vectors are now", allow_reread=False)[0]
+        if not f:
+            return None
+        for i in range(4):
+            self.readline() # skip 4 lines
+        C = []
+        for i in range(3):
+            line = self.readline()
+            v = line.split()
+            C.append(v[:3]) # direct lattice vectors
+        # read a position-force table
+        f = self.step_to("TOTAL-FORCE (eV/Angst)", allow_reread=False)[0]
+        if not f:
+            return None
+        self.readline() # -----
+        P, F = [], []
+        line = self.readline()
+        while "----" not in line:
             v = line.split()
-            E["total"] = float(v[4])
-            E["sigma0"] = float(v[-1])
-            return E
-
-        itt = iter(self)
-        E = []
-        e = readE(itt)
-        while e is not None:
-            E.append(e)
-            e = readE(itt)
-        try:
-            # this just puts the job_completed flag. But otherwise not used
-            self.cpu_time()
-        except Exception:
-            pass
-
-        if all:
-            return E
-        if len(E) > 0:
-            return E[-1]
-        return None
+            # positions and forces
+            P.append(v[:3])
+            F.append(v[3:6])
+            line = self.readline()
+        step = PropertyDict()
+        step.cell = np.array(C, dtype=np.float64)
+        step.xyz = np.array(P, dtype=np.float64)
+        step.force = np.array(F, dtype=np.float64)
+        return step
```

Comment:
perhaps a `postprocess` argument to `SileBinder` to unpack the Property dict?
```python
def unpack2collection(steps):
    ret = dict()
    for step in steps:
        for key, val in step.items()):
            ret[key].append(val) # won't work as is not list-initialized, but...
    prop = PropertyDict()
    for key, steps in ret.items():
        prop[key] = Collection(steps)
    return prop
```
we really need to discuss how we should deal with multiple return values. and arguments for such

Reply:
I've addressed all comments except this one as I am not sure we want/need to unpack. For instance, the current behaviour seems logical to me:
```python
traj = stdoutSileVASP(f).read_trajectory[:3]()
for i in range(3):
    print(traj[i].force)
# or
for step in traj:
    print(step.force)
```
You would prefer this?
```python
traj = stdoutSileVASP(f).read_trajectory[:3]()
for i in range(3):
    print(traj.force[i])
# or
for F in traj.force:
    print(F)
```